staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1179,6 +1207,23 @@ func (c *cacheWatcher) nextBookmarkTime(now time.Time) (time.Time, bool) {
 	return c.deadline.Add(-2 * time.Second), true
 }
 
+func getEventObject(object runtime.Object) runtime.Object {
+	if _, ok := object.(*cachingObject); ok {
```

Comment:
is there a reason this is checking the implementation type and not the CacheableObject interface?